File: oojs-orange-tree-challenge-master/spec/orange_tree_spec.js
describe("an orange tree", function() {
  var tree;

  beforeEach(function() {
    tree = new OrangeTree();
  });

  it("has an age");
  it("has a height");
  it("has a collection of oranges");

  describe("reporting whether it's mature", function() {
    it("is mature if it has reached fruit-bearing age");
    it("is not mature if it has not reached fruit-bearing age");
  });

  describe("reporting whether it's dead", function() {
    it("is dead if it's reached the maximum age for an orange tree");
    it("is not dead if it's not reached the maximum age for an orange tree");
  });

  describe("reporting whether it has oranges", function() {
    it("has oranges if it's collection of oranges is not empty");
    it("has no oranges if it's collection of oranges is empty");
  });

  describe("passing a growing season", function() {
    describe("when it's alive", function() {
      it("gets older");

      describe("when the tree is shorter than the maximum height for an orange tree", function() {
        it("grows taller");
      });

      describe("when the tree has reached the maximum height for an orange tree", function() {
        it("does not grow");
      });

      describe("when it's mature", function() {
        it("produces oranges");
      });

      describe("when it's not mature", function() {
        it("does not produce fruit");
      });
    });

    describe("when it's dead", function() {
      it("does not get older");
      it("does not grow");
      it("does not produce fruit");
    });
  });

  describe("picking an orange", function() {
    describe("when the tree has oranges", function() {
      it("returns one of its oranges");
      it("no longer has the returned orange in its collection of oranges");
    });

    describe("when the tree has no oranges", function() {
      it("returns undefined");
    });
  });
});

File: orange-tree-1-just-oranges-challenge-master/spec/orange_tree_spec.rb
require_relative "../orange"
require_relative "../orange_tree"


describe OrangeTree do
  let(:tree) { OrangeTree.new }

  #You'll need to switch `pending` to `describe` when you're ready
  # to implement each set of tests.

  pending "#age" do
    it "has an age" do
    end
  end

  pending "#height" do
    it "has a height" do
    end
  end


  pending "#pass_growing_season" do
    it "should change the age" do
      # This should be more explicit. How much should the tree age by?
      # https://www.relishapp.com/rspec/rspec-expectations/v/2-0/docs/matchers/expect-change
      expect {tree.pass_growing_season}.to change {tree.age}
    end

    it "should make the tree grow" do
      #This should be more explicit. How much should the tree grow?
      expect {tree.pass_growing_season}.to change{tree.height}
    end

    context "the tree is old enough to bear fruit" do
      it "should cause the tree to produce oranges" do
      end
    end
  end

  pending "#mature?" do
    it "returns true if tree is old enough to bear fruit" do
    end

    it "returns false if tree is not old enough to bear fruit" do
    end
  end

  pending "#dead?" do
    it "should return false for an alive tree" do
    end

    it "should return true for a dead tree" do
    end
  end

  pending '#has_oranges?' do
    it 'should return true if oranges are on the tree' do
    end

    it "should return false if the tree has no oranges" do
    end
  end

  pending "#pick_an_orange" do
    it "should return an orange from the tree" do
    end

    it "the returned orange should no longer be on the tree" do
    end

    it "should raise an error if the tree has no oranges" do
    end
  end
end
